### PR TITLE
doc: add a nice explainer about how to update PR branches

### PR DESCRIPTION
**Pull request by niftynei**
This is good info to have, that I often forget and have to spend time looking for. It'd be nice to have it documented in the project somewhere so people new the project (or new to maintaining maybe?) know where to find it.

Open to suggestions of other places to put this; totally understandable if it's not useful to have in this repo, particularly.

### CODE REVIEW

**Comment by cdecker** [comment]
Rebased on top of `master`

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/pull/6252/commits/e0540b516e3247601a2ef324d738ce49bd003e53

But I had a comment; it is not a blocking one because git has a lot of different ways to do the same things, and I may misunderstood how to use your suggestion

**Comment by rustyrussell** [comment]
This is easier with the github cmdline tools BTW!

But it needs moving to the updated docs...

**Comment by niftynei** [comment]
> This is easier with the github cmdline tools BTW!

Hmm. Do most people install those? I can add a second entry with info about it.

**Comment by vincenzopalazzo** [comment]
>Hmm. Do most people install those? 

I never use it but I guess a common things for people that are  new to git, looks like a good twitter pool :)

File: doc/HACKING.md [hunk 1]
```diff
@@ -322,6 +322,61 @@ directory `../bolts` and a recent git version.
...
 e.g., `make extract-bolt-csv BOLTDIR=../bolts BOLTVERSION=ee76043271f79f45b3392e629fd35e47f1268dc8`
 
+
+Pushing Up Changes to PR Branches
+---------------------------------
+
+If you want to pull down and run changes to a PR branch, you can use the convenient
+pr/<pr#> branch tags to do this.
+
+	git checkout pr/<pr#>
```
[the remark this answers is not in the document]
Comment (rustyrussell) [reply doc/HACKING.md:332]: I think this needs the .github/config magic of:

```
[remote "origin"]
    fetch = +refs/pull/*/head:refs/remotes/origin/pr/*
```

**Review by rustyrussell — CHANGES_REQUESTED** [review]

**Comment by rustyrussell** [comment]
Note: doc moved and updated in doc/contribute-to-core-lightning/contributor-workflow.md

File: doc/HACKING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (niftynei) [reply doc/HACKING.md:332]: ah yeah, i'll add that!

**Comment by nepet** [comment]
Removed the milestone as this PR seems to be stuck.

**Comment by niftynei** [comment]
Added requested comment about the `pr/#` folder and moved to new docs location.

**Comment by cdecker** [comment]
I rebased this on top of `master`, which was a bit complicated since there was a rename of the file inbetween. Sorry for dropping your authorship @niftynei but I couldn't think of a simpler way than to hard reset and reapply your diff on the new filename instead.

This will be merged as soon as the CI passes :fireworks:

**Pull request merged**